src/lib/sign.c

```diff
@@ -149,6 +154,42 @@ static CK_RV update_pss_sig_state(token *tok, tobject *tobj) {
     return rv;
 }
 
+static CK_RV policy_is_satisfied(tpm_ctx *tctx, tobject *tobj, uint32_t handle) {
+
+    CK_ATTRIBUTE_PTR policy_attr = attr_get_attribute_by_type(tobj->attrs, CKA_TPM2_POLICY_JSON);
+    if (!policy_attr) {
+        /* nonexistent policy is always satisfied */
+        return CKR_OK;
+    }
+
+#ifndef HAVE_POLICY
+    UNUSED(tctx);
+    UNUSED(handle);
+    /* the policy is considered to be satisfied, but will warn about it */
+    LOGW("Found a policy, but support for enforcing it wasn't compiled in");
+    return CKR_OK;
+#else
```

Comment:
I would create 2 versions of this function and ifdef between the two and avoid even looking for the policy on non-policy enabled builds.